src/translators/rust/pure/cargo-lock/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  externals,
+  translatorName,
+  utils,
+  lib,
+  runCommandNoCC,
+  ...
+}:
+
+let
+  translate =
+    {
+      inputPaths,
+      ...
+    }:
+    let
+      cargoLock = builtins.fromTOML (builtins.readFile (builtins.elemAt inputPaths 0));
+      parsed = builtins.listToAttrs (map (p: lib.nameValuePair p.name p) cargoLock.package);
```

Comment:
if you are interested in some of the parsing that nixpkgs does on the cargolock, there was recently a PR about some improvements to it. You might be interested.
https://github.com/NixOS/nixpkgs/pull/137303/files